src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,29 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.Arrays;
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_YEARS_LIVED = 10;
+    private static final int MIN_AGE = 35;
+    private static final int FIRST_YEAR_INDEX = 0;
+    private static final int SECOND_YEAR_INDEX = 1;
+    private static final String NATIONALITY = "Ukrainian";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        int[] yearsLived = Arrays.stream(candidate
+                        .getPeriodsInUkr()
+                        .split("-"))
```

Comment:
All magic numbers, strings, and symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
```suggestion
                        .split(DELIMITER))
```